CMakeLists.txt

```diff
@@ -513,24 +503,53 @@ CONFIGURE_FILE(
   )
 
 
+# Check if we want to precompile header files. This speeds up compile time,
+# but can fail on some machines with old cmake, therefore it is disabled by
+# default. Also cotire currently has issues with cmake generator expressions
+# used in deal.II for cmake version 3.9 and above. Disable cotire for cmake >=
+# 3.9. Starting with cmake 3.15 there is native support inside cmake and we can
+# use that.
+IF(CMAKE_VERSION VERSION_LESS 3.9)
+  SET(ASPECT_PRECOMPILE_HEADERS OFF CACHE BOOL "Precompile external header files using cotire to speedup compile time. Currently only supported for cmake versions older than 3.9 or newer than 3.15.")
+ELSEIF(CMAKE_VERSION VERSION_GREATER 3.15)
+  SET(ASPECT_PRECOMPILE_HEADERS ON CACHE BOOL "Precompile external header files to speedup compile time. Currently only supported for cmake versions older than 3.9 or newer than 3.9.")
+ENDIF()
 
 IF (ASPECT_PRECOMPILE_HEADERS)
-  MESSAGE(STATUS "Enabling cotire to precompile external header files.")
-  INCLUDE(cmake/cotire)
-
-  # Set some properties that are necessary for the header precompilations 
-  # and unity build to pass. We need to exclude some headers that
-  # frequently cause problems, and undefine some macros between source 
-  # files that would otherwise interfere with each other.
-  SET_TARGET_PROPERTIES (aspect PROPERTIES
-      COTIRE_PREFIX_HEADER_IGNORE_PATH
-      "/usr/include/;${CMAKE_SOURCE_DIR};${CMAKE_BINARY_DIR}")
-
-  # This line activates the cotire module for the aspect target, 
-  # and therefore the whole precompilation
-  cotire(aspect)
+  IF (CMAKE_VERSION VERSION_LESS 3.9)
+    MESSAGE(STATUS "Enabling cotire to precompile external header files.")
```

Comment:
I would suggest we get rid of cotire entirely. It is too complicated to make sure all 3 variants work (and I don't see how we can test this on jenkins). We can delay this until we bump to deal.II 9.2 if you prefer.